crates/ruff_server/src/session/settings.rs

```diff
@@ -418,6 +414,37 @@ impl ResolvedClientSettings {
                     ConfigurationPreference::EditorFirst,
                 ),
             },
+        };
+
+        if contains_invalid_settings {
+            show_err_msg!(
+                "Ruff received invalid client settings. Refer to the logs for more information."
+            );
+        }
+
+        settings
+    }
+
+    fn resolve_rules(
+        rules: &[String],
+        key: RuleSelectorKey,
+        contains_invalid_settings: &mut bool,
+    ) -> Option<Vec<RuleSelector>> {
+        let (mut known, mut unknown) = (vec![], vec![]);
+        for rule in rules {
+            match RuleSelector::from_str(rule) {
+                Ok(selector) => known.push(selector),
+                Err(ParseError::Unknown(_)) => unknown.push(rule),
+            }
+        }
+        if !unknown.is_empty() {
+            *contains_invalid_settings = true;
+            tracing::error!("Unknown rule selectors found in `{key}`: {unknown:?}");
```

Comment:
Is this an error message that we could log from within Ruff itself or does it only apply to a subset of settings?

Reply:
I don't think Ruff has a tracing system configured as it uses a basic [`log`](https://docs.rs/log/latest/log/) crate. So, those logs might not show up in the editor log channel

Reply:
Tracing seems to work. At least, I can' see logs for the `format_path` spans

```
[2025-02-27][09:13:09][ruff::commands::format][DEBUG] format_path; path=/Users/micha/astral/ruff/scripts/knot_benchmark/src/benchmark/__init__.py
[2025-02-27][09:13:09][ruff::commands::format][DEBUG] format_path; path=/Users/micha/astral/ruff/scripts/knot_benchmark/src/benchmark/run.py
[2025-02-27][09:13:09][ruff::commands::format][DEBUG] format_path; path=/Users/micha/astral/ruff/scripts/knot_benchmark/src/benchmark/projects.py
[2025-02-27][09:13:09][ruff::commands::format][DEBUG] format_path; path=/Users/micha/astral/ruff/scripts/knot_benchmark/src/benchmark/cases.py
```

Reply:
This would also mean that a message can only contain a single rule, so there'll be `n` number of messages where `n` are the total number of unknown rule selectors.

Reply:
Oh, wait, you meant to put this directly in the resolution of `lint.select` inside Ruff. Let me check

Reply:
Okay, thanks for checking.

I'm mainly asking because it would be nice to have less server specific logging. 

It might be worth exploring if we should change all ruff `debug`, and `trace` usages to use tracing instead of `log` so that they'd be visible in the server too. But that's something for another day.